sklearn/neural_network/tests/test_mlp.py

```diff
@@ -266,6 +266,32 @@ def test_lbfgs_regression():
             # Non linear models perform much better than linear bottleneck:
             assert_greater(mlp.score(X, y), 0.95)
 
+def test_lbfgs_maxfun():
+    # Test lbfgs parameter max_fun
+    # It should independently limit the 
+    # number of iterations for lbfgs
+    max_fun = 10
+    expected_num_iter = 10 
+    for X, y in classification_datasets:
+        X_train = X[:150]
+        y_train = y[:150]
+        X_test = X[150:]
+
+
+        for activation in ACTIVATION_TYPES:
+            mlp = MLPClassifier(solver='lbfgs', hidden_layer_sizes=50,
+                                max_iter=150, max_fun=max_fun, shuffle=True, 
+                                random_state=1, activation=activation)
+            mlp.fit(X_train, y_train)
+            assert_greater(expected_num_iter, mlp.n_iter_)
```

Comment:
add `# regression tests:`

Reply:
I think it would be clearer here if you use `max_fun` rather than `expected_n_iter`. I think the idea that `max_fun` bounds the number of iterations strictly from above is quite clear then.